src/GatewayAPI/Database/TransactionQuerier.cs

```diff
@@ -332,37 +344,113 @@ private async Task<List<long>> GetAccountTransactionStateVersions(AccountTransac
 
     private async Task<List<Gateway.TransactionInfo>> GetTransactions(List<long> transactionStateVersions)
     {
-        var transactionWithOperationGroups = await _dbContext.LedgerTransactions
-            .Where(t => transactionStateVersions.Contains(t.ResultantStateVersion))
-            .Include(t => t.SubstantiveOperationGroups)
-            .ThenInclude(op => op.InferredAction!.Resource)
-            .Include(t => t.SubstantiveOperationGroups) // https://stackoverflow.com/a/50898208
-            .ThenInclude(op => op.InferredAction!.Validator)
-            .Include(t => t.SubstantiveOperationGroups) // https://stackoverflow.com/a/50898208
-            .ThenInclude(op => op.InferredAction!.FromAccount)
-            .Include(t => t.SubstantiveOperationGroups) // https://stackoverflow.com/a/50898208
-            .ThenInclude(op => op.InferredAction!.ToAccount)
-            .Include(t => t.RawTransaction)
-            .OrderByDescending(lt => lt.ResultantStateVersion)
-            .AsSplitQuery() // See https://docs.microsoft.com/en-us/ef/core/querying/single-split-queries
-            .ToListAsync();
+        async Task<List<LedgerTransaction>> LoadLedgerTransactions(List<long> ids)
+        {
+            await using var localCtx = await _dbContextFactory.CreateDbContextAsync();
+
+            // we're using explicit transaction here to ensure that:
```

Comment:
Added some (minor) style comments on slack

Reply:
Could we please try to keep reviews limited to GH? ;-)